### ISSUE TITLE
Checks/tasks required before "release"

### ISSUE THREAD

**Issue by cpavlina** [cpavlina/kicad-schlib#7]
This is a "living library", so there are no true releases, but I'd like to concentrate on some cleanup tasks that need to be performed to get it into a "release ready" status. Once this is done I may evangelize a bit more for this library :)

This issue is to collect a list of such tasks. So far:

- Simple tasks:
  - Standard field size: all must be 50 mil
  - Field placement cleanup: arrange even hidden fields neatly so they're readable in libedit
  - Pin size: meet the requirements in `DESIGN.md`

- Specific library cleanup tasks:
  - Check SWDIO/SWCLK on all ATSAMD* parts. Make sure these functions are assigned to the correct pins, and make sure SWDIO is positioned above SWCLK on the symbol to match CONN-CORTEX-DEBUG.
  - Check voltage regulator pin types. In particular, some three-terminal regulators have ADJ set to Power Output (a vestige from this having been a GND pin that was edited). This should be Passive.

- Large library cleanups. **A separate issue will be created for each of these**
  - Finish logic library rewrite
  - Start and finish connector library rewrite
  - Power library rewrite
  - Passive symbol type cleanup - IEC vs ANSI, etc

- Policy to-do:
  - Standardize styling of microcontrollers. How are pins named (just names like `PA23`, or are peripheral functions included?), which pins are pulled out into isolated positions (programming? oscillator? voltage reference?), etc.
  - Write contributor guide

**Comment by cpavlina** [comment]
As some of the new libraries are going to use the footprint filter list, I've created an issue in KiCad's tracker for making that a bit more usable in the new footprint field-based workflow:

[Wish: use fp filter list in footprint selector](https://bugs.launchpad.net/bugs/1660131)

I'll probably implement that myself if nobody else does. I'm not sure how much code this single-part footprint selector shares with cvpcb, it may be easy or difficult to include this existing cvpcb functionality in it.

**Comment by iromero91** [comment]
For the microcontroller symbols, we just need to accept it is absolutely impossible to please everyone, specially with modern huge microcontrollers where each pin can have like 10 different functions. I for one absolutely despise symbols that just go and list every single function for every pin separated by slashes, With the new format i would hope it would be easier to take base symbols and edit them, like renaming pins, so for generic symbols in the library i propose this:

* For single function pins, like power, reset-only, boot0, etc, name them as they are and arrange them in a way that makes sense for their usage.

* For GPIOs that have an unique, "important" secondary function that most people expect to be using, name them by the GPIO name slash the secondary function, like "RA6/~RESET". For example, Reset, crystal oscillators, programming/debug interface, etc. Pins that are not plain GPIOs on reset would be there too, also if the part has system memory with a bootloader (like STM32, name the pins like uarts and USB peripherals used by it with the same scheme)

* The rest of the GPIOs should be just named by the plain GPIO name, even though it could have 30 alternate functions. At this point i think it's easier for the user to just modify the symbol to select exactly which of the 30 they are actually using in their design.

**Comment by cpavlina** [comment]
First priority now is to run through the first section (simple tasks) one library at a time; I am doing this now.
